src/classes/CW_AudienceCSVUploadControllerTest.cls

```diff
@@ -0,0 +1,397 @@
+		insert sch;
+		
+		Test.setCurrentPage(Page.CW_AudienceCSVUpload);
+		Apexpages.currentPage().getParameters().put(ConstantUtil.PAGEPARAM_CAMPAIGNID,camp.Id);
+		
+		Test.startTest();
+		CW_AudienceCSVUploadController uc = new CW_AudienceCSVUploadController();
+		uc.selectedSchId = sch.id;
+		uc.csvFile = Blob.valueOf(groupCSVString);
+		uc.filename='Somefile.csv';
+		uc.uploadCSV();
+		//uc.returnToAudiencePage();
+		Test.stopTest();
+		
+		// 6 new accounts were added
+		system.assertEquals(6,[select count() from Contact where Account_Name__c != :null]);
+		// 5 new groupss were added
+		system.assertEquals(5,[select count() from CW_Audience_Group__c where Campaign__c = :camp.Id]);
+		// 2 members where added to group1
+		system.assertEquals(2,[select count() from CW_Audience_Group_Member__c where Audience_Group_LU__r.Name = :group1]);
+		// 1 members where added to group2
+		system.assertEquals(1,[select count() from CW_Audience_Group_Member__c where Audience_Group_LU__r.Name = :group2]);
+		// 1 members where added to group3
+		system.assertEquals(1,[select count() from CW_Audience_Group_Member__c where Audience_Group_LU__r.Name = :group3]);
+		// 1 members where added to group4
+		system.assertEquals(1,[select count() from CW_Audience_Group_Member__c where Audience_Group_LU__r.Name = :group4]);
+		// 1 members where added to group5
+		system.assertEquals(1,[select count() from CW_Audience_Group_Member__c where Audience_Group_LU__r.Name = :group5]);
+		
+	}
+	
+	static testMethod void parseGroupsAndMembersBatch()
+	{
+		CW_Campaign__c camp = TestDataUtil.createCampaign();
+		insert camp;
+		CW_Schedule__c sch = TestDataUtil.createSchedule(camp);
+		insert sch;
+		
+		String memebrsCSVString = 'Group Name,Account Name' + CSVReader.lineTerminator;
+		
+		for(Integer i=1; i<=1000; i++)
+		{
+			
+			String playerId = 'shl' + i;
+			memebrsCSVString += (i <= 500 ? group1 : group2) + ',' + playerId + CSVReader.lineTerminator;
+		}
+		memebrsCSVString = memebrsCSVString.removeEnd(CSVReader.lineTerminator);
+		
+		Test.setCurrentPage(Page.CW_AudienceCSVUpload);
+		Apexpages.currentPage().getParameters().put(ConstantUtil.PAGEPARAM_CAMPAIGNID,camp.Id);
+		
+		Test.startTest();
+		CW_AudienceCSVUploadController uc = new CW_AudienceCSVUploadController();
+		uc.selectedSchId = sch.Id;
+		uc.csvFile = Blob.valueOf(memebrsCSVString);
+		uc.filename='Somefile.csv';
+		uc.uploadCSV();
+		uc.loadBatchJobProgressPage();
+		Test.stopTest();
+		system.assertEquals(1000,[select count() from Contact where Account_Name__c != :null]);
+		// 2 new groups were added
+		system.assertEquals(2,[select count() from CW_Audience_Group__c where Campaign__c = :camp.Id]);
+		system.assertEquals(500,[select count() from CW_Audience_Group_Member__c where Audience_Group_LU__r.Name = :group1]);
+		system.assertEquals(500,[select count() from CW_Audience_Group_Member__c where Audience_Group_LU__r.Name = :group2]);
+	}
+	
+	static testMethod void parseFaultyGroups()
```

Comment:
Style issue: test methods should be annotated with @IsTest for better readibility